wandb/util.py

```diff
@@ -1182,20 +1130,20 @@ def auto_project_name(program):
     root_dir = to_native_slash_path(root_dir)
     repo_name = os.path.basename(root_dir)
     if program is None:
-        return repo_name
+        return str(repo_name)
     if not os.path.isabs(program):
         program = os.path.join(os.curdir, program)
     prog_dir = os.path.dirname(os.path.abspath(program))
     if not prog_dir.startswith(root_dir):
-        return repo_name
+        return str(repo_name)
     project = repo_name
     sub_path = os.path.relpath(prog_dir, root_dir)
     if sub_path != ".":
         project += "-" + sub_path
-    return project.replace(os.sep, "_")
+    return str(project.replace(os.sep, "_"))
```

Comment:
why the strs in this function... werent they already strings?   i thought os.path. methods return strings

Reply:
mostly just curious.

Reply:
sigh... wondering that myself. this is a trick that made `mypy` happy..

Reply:
works for me, good to keep mypy happy.